sonar-bukkit/src/main/java/xyz/jonesdev/sonar/bukkit/fallback/FallbackBukkitInjector.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2023-2024 Sonar Contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package xyz.jonesdev.sonar.bukkit.fallback;
+
+import io.netty.channel.*;
+import lombok.experimental.UtilityClass;
+import org.bukkit.Bukkit;
+import org.jetbrains.annotations.NotNull;
+import xyz.jonesdev.sonar.api.ReflectiveOperationException;
+import xyz.jonesdev.sonar.api.Sonar;
+import xyz.jonesdev.sonar.common.fallback.netty.FallbackInjectedChannelInitializer;
+
+import java.lang.reflect.Field;
+import java.util.List;
+
+import static xyz.jonesdev.sonar.api.fallback.FallbackPipelines.FALLBACK_PACKET_HANDLER;
+
+// Check out these links if you want to see some more magic
+// https://github.com/retrooper/packetevents/blob/2.0/spigot/src/main/java/io/github/retrooper/packetevents/util/SpigotReflectionUtil.java
+// https://github.com/dmulloy2/ProtocolLib/blob/master/TinyProtocol/src/main/java/com/comphenix/tinyprotocol/TinyProtocol.java
+@UtilityClass
+public class FallbackBukkitInjector {
+  private final String MODIFIED_PACKAGE_NAME;
+  private final String LEGACY_NMS_PACKAGE;
+  private final String OBC_PACKAGE;
+
+  private final BukkitServerVersion SERVER_VERSION;
+  private final boolean USES_LEGACY_PACKAGING, OBFUSCATED;
+
+  private final Class<?> MINECRAFT_SERVER_CLASS;
+  private final Class<?> CRAFTBUKKIT_SERVER_CLASS;
+  private final Class<?> SERVER_CONNECTION_CLASS;
+
+  private Object MINECRAFT_SERVER_INSTANCE;
+  private final Object MINECRAFT_SERVER_CONNECTION_INSTANCE;
+
+  static {
+    final String bukkitPackage = Bukkit.getServer().getClass().getPackage().getName();
+
+    MODIFIED_PACKAGE_NAME = bukkitPackage.split("\\.")[3];
```

Comment:
You may need to handle with https://forums.papermc.io/threads/important-dev-psa-future-removal-of-cb-package-relocation.1106/

Reply:
We're using 1.8.8 as the Bukkit API version. It's easier to simply remove the version identifier at the end

Reply:
![2024-07-14_16 52 35](https://github.com/user-attachments/assets/c8587a3e-966e-4764-8861-144fac25e4d2)

Verification works fine. No errors in console. The only problem now is the ViaVersion stuff.

Reply:
I posted a little late.I posted a little late. But I thinking if servers like Spigot have changed the package name as well.

Reply:
> I posted a little late.I posted a little late. But I thinking if servers like Spigot have changed the package name as well.

Sonar only needs the OBC (org.bukkit.craftbukkit) and NMS (net.minecraft.server) packages. It doesn't directly use any APIs. It works fine how it is implemented right now, so I think we can mark this conversation as resolved. If you have any concerns, you can ping me on the #development channel on Discord and I'll take a deeper look into the matter, however, it works now, I don't know what else check for.